velox/row/UnsafeRowFast.cpp

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/row/UnsafeRowFast.h"
+
+namespace facebook::velox::row {
+
+namespace {
+static const int32_t kFieldWidth = 8;
+
+int32_t alignBits(int32_t numBits) {
+  return bits::nwords(numBits) * 8;
+}
+
+int32_t alignBytes(int32_t numBytes) {
+  return bits::roundUp(numBytes, 8);
+}
+} // namespace
+
+// static
+std::optional<int32_t> UnsafeRowFast::fixedRowSize(const RowTypePtr& rowType) {
+  for (const auto& child : rowType->children()) {
+    if (!child->isFixedWidth()) {
+      return std::nullopt;
```

Comment:
Since the return type is std::optional, is it better to use `std::optional<int32_t>` to signify empty value?

Reply:
@vermapratyush 

https://en.cppreference.com/w/cpp/utility/optional/nullopt

`std::nullopt is a constant of type [std::nullopt_t](https://en.cppreference.com/w/cpp/utility/optional/nullopt_t) that is used to indicate optional type with uninitialized state.`